ecommers/lib/core/provider_models/products_grid_provder_model.dart

```diff
@@ -1,57 +1,84 @@
 import 'package:async/async.dart';
 import 'package:flutter/material.dart';
 
+import 'package:ecommers/core/mixins/items_loading_notifier.dart';
+
 import 'package:ecommers/core/common/index.dart';
 import 'package:ecommers/core/models/data_models/index.dart';
 import 'package:ecommers/core/models/sort_type.dart';
 import 'package:ecommers/core/provider_models/provider_model_base.dart';
-import 'package:ecommers/core/services/index.dart';
+import 'package:ecommers/core/services/index.dart' as services;
+
+class ProductsGridProviderModel extends ProviderModelBase
+    with ItemsLoadingNotifier {
+  final Categories _categoryType;
+  final String _subCategory;
+  final SortType _sortType;
+  String _searchQuery;
 
-class ProductsGridProviderModel extends ProviderModelBase {
   CancelableOperation<List<Product>> _fetchProductsOperation;
-  List<Product> _products;
 
+  List<Product> _products;
   List<Product> get products => _products;
 
   ProductsGridProviderModel({
     @required BuildContext context,
-  }) : super(context);
-
-  Future _fetchProducts(
     Categories categoryType,
     String subCategory,
-    String searchQuery,
     SortType sortType,
-  ) async {
+  })  : _categoryType = categoryType,
+        _subCategory = subCategory,
+        _sortType = sortType,
+        super(context);
+
+  Future updateProducts(String searchQuery) async {
+    if (_fetchProductsOperation != null &&
+        !_fetchProductsOperation.isCompleted) {
+      _fetchProductsOperation.cancel();
+    }
+
+    _searchQuery = searchQuery;
+
+    _products = await _fetchProducts();
+  }
+
+  @override
+  Future loadMoreProducts() async {
+    isitemsLoading = true;
+
+    final products = await paginator.loadNextPage(_fetchMoreProducts);
```

Comment:
why don't  you write services.paginator?

Reply:
because it's not a services property:)